distil/primitives/ranked_linear_svc.py

```diff
@@ -191,33 +192,56 @@ def produce(
             inputs = normalize(inputs)
         result = self._model.predict(inputs)
         result_df: container.DataFrame = None
+
         if self._binary:
             if self.hyperparams["confidences"]:
+                # If we're generating confidences, check to see where or not we calibrate
+                # to probabilities, or just return the raw decision function values.
                 if self.hyperparams["calibrate"]:
                     cccv = CalibratedClassifierCV(self._model, cv="prefit")
                     cccv.fit(inputs, result)
                     confidences = cccv.predict_proba(inputs)[:, 1]
                 else:
                     confidences = self._model.decision_function(inputs)
+
+                # Generate ranks if required, otherwise we just include the confidence / decision
+                # function values.
                 if self.hyperparams["rank_confidences"]:
-                    confidences = rankdata(confidences)
-                result_df = container.DataFrame(
-                    {self._target_cols[0]: result, "confidence": confidences},
-                    generate_metadata=True,
-                )
+                    ranks = rankdata(confidences)
+                    # ranks = 1.0 - (
+                    #     (ranks - np.min(ranks)) / (np.max(ranks) - np.min(ranks))
+                    # )
+                    # result_df = container.DataFrame(
+                    #     {
+                    #         self._target_cols[0]: result,
+                    #         "confidence": confidences,
+                    #         "rank": ranks,
```

Comment:
Is this code suppose to be commented out? I don't see where the `ranks` value gets used. This branch of the code won't generate a `result_df`.

Reply:
No, part of that should have been uncommented.  I'll fix it now.